analyses/immune-deconv/02-summary-plots.R

```diff
@@ -64,13 +67,39 @@ create.heatmap <- function(deconv.method, title) {
     spread(key = label, value = mean) %>% 
     column_to_rownames('cell_type')
   
-  # remove rows with all zeros (not allowed because we are scaling by row)
-  deconv.method <- deconv.method[apply(deconv.method, 1, function(x) !all(x==0)),]
-  
-  title <- paste0(title,"\nAverage immune scores normalized by rows")
-  pheatmap(mat = t(deconv.method), fontsize = 10, 
-           scale = "column", angle_col = 45,
-           main = title, annotation_legend = T, cellwidth = 15, cellheight = 15)
+  # plot non-brain and brain tumors separately
+  pdf(file = fileout, width = 13, height = 8)
+  # non-brain tumors
+  mat <- deconv.method %>% 
+    select(grep(paste0(non.brain.tumors, collapse="|"), colnames(deconv.method), value = TRUE)) 
```

Comment:
I believe you could use [`tidyselect::matches`](https://www.rdocumentation.org/packages/tidyselect/versions/1.0.0/topics/select_helpers) here: 
```suggestion
    select(matches(paste0(non.brain.tumors, collapse="|"))) 
```
(untested)